crates/chain/src/spk_txout_index.rs

```diff
@@ -334,4 +343,33 @@ impl<I: Clone + Ord> SpkTxOutIndex<I> {
             .any(|output| self.spk_indices.contains_key(&output.script_pubkey));
         input_matches || output_matches
     }
+
+    /// Return `TotalSentReceived` for a `Transaction` in relation to the `SpkTxOutIndex`.
```

Comment:
Isn't this meant to replace `sent_and_received`? (it's still here).
After seeing this I think the tuple `(u64, u64)` as it was before since  has the order of the returned value is in the name `total_sent_received` which implies `sent` is being returned before received. Not a major concern so feel free to go with whatever feels right.

In any case please just rename the existing function `sent_and_received` and change the output type (if you still want a custom type).